extensions/amp-fx-scroll-sync/0.1/scroll-sync-rotate-effect.js

```diff
@@ -14,45 +14,64 @@
  * limitations under the License.
  */
 
- import {ScrollSyncEffect} from './scroll-sync-effect';
- import {setStyles} from '../../../src/style';
-
- export class ScrollSyncRotateEffect extends ScrollSyncEffect {
-   constructor(element, config) {
-     super(element);
-     this.scrollMin_ = config['starting-position'];
-     this.scrollMax_ = config['ending-position'];
-     this.rotateAngle_ = config['rotate-angle'];
-   }
-
-   /** @override */
-   isDirectional() {
-     return false;
-   }
-
-   /** @override */
-   measure() {
-
-   }
-
-   /** @override */
-   requestMeasure() {
-
-   }
-
-   /** @override */
-   getScrollMin() {
-     return this.scrollMin_;
-   }
-
-   /** @override */
-   getScrollMax() {
-     return this.scrollMax_;
-   }
-
-   /** @override */
-   transition(position) {
-     const angle = position * this.rotateAngle_;
-     setStyles(this.element, {'transform': `rotate(${angle}deg)`});
-   }
- }
+import {ScrollSyncEffect} from './scroll-sync-effect';
+import {setStyles} from '../../../src/style';
+import {getLengthNumeral} from '../../../src/layout';
+import {user} from '../../../src/log';
+
+export class ScrollSyncRotateEffect extends ScrollSyncEffect {
+  constructor(element) {
+    super(element);
+    const rotateEffectElements = element.querySelectorAll('[type="rotate"]');
```

Comment:
Yeah, we might allow having the same effect multiple times. Doing that might create effect conflict, which we are not handling it right currently. I'll create a TODO for allowing multiple same effects and we can think about how to deal with that later.